File: Data Structures & Algorithms/two-integer-sum/submission-2.go
func twoSum(nums []int, target int) []int {
	j, i := 0, 1
    for j < len(nums) {
		tmp := nums[j] + nums[i]
		if tmp == target && j != i {
			return []int{j, i}
		}
		i++
		if i == len(nums) {
			j++
			i = j + 1
		}
	}
	return []int{}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        j = i + 1
        while i < len(numbers):
            p1 = numbers[i]
            p2 = numbers[j]
            if p1 + p2 == target:
                return [i+1,j+1]
            if j >= len(numbers)-1:
                i+=1
                j = i + 1
            else:
                j += 1
        
        return [0,0]